### PR TITLE
Feature/select

### LINKED ISSUE #10: Add nicer selects
https://github.com/monterail/vue-multiselect

### PR DESCRIPTION
**Pull request by kjugi**
Delete select component
Add vue-multiselect lib & styles
Add vue-multiselect selects instead normal one

Closes #10

### CODE REVIEW

File: view/frontend/web/js/index.vue [hunk 1]
```diff
@@ -7,6 +7,8 @@
   </div>
 </template>
 
+<style src="./assets/vue-multiselect.min.css"></style>
+
```
Comment (szafran89) [inline view/frontend/web/js/index.vue]: Always add styles on the bottom of SFC

File: view/frontend/web/js/components/steps/TheShippingStep.vue [hunk 2]
```diff
@@ -281,6 +295,16 @@ export default {
...
+  },
+  watch: {
+    countryWatcher (newCountry) {
+      this.address.country_id = newCountry.value
+      this.regionWatcher = ''
+      this.onCountryChange()
+    },
+    regionWatcher (newRegion) {
+      this.address.region_id = newRegion.value
+    }
```
Comment (szafran89) [inline view/frontend/web/js/components/steps/TheShippingStep.vue]: I think you don't need to use watchers here. IMO regular component data is enough.

File: view/frontend/web/js/components/steps/ThePaymentStep.vue [hunk 3]
```diff
@@ -253,6 +267,16 @@ export default {
...
+  },
+  watch: {
+    countryWatcher (newCountry) {
+      this.address.country_id = newCountry.value
+      this.regionWatcher = ''
+      this.onCountryChange()
+    },
+    regionWatcher (newRegion) {
+      this.address.region_id = newRegion.value
+    }
```
Comment (szafran89) [inline view/frontend/web/js/components/steps/ThePaymentStep.vue]: I think you don't need to use watchers here. IMO regular component data is enough.

File: view/frontend/web/js/assets/vue-multiselect.min.css [hunk 4]
```diff
@@ -0,0 +1 @@
...-events:none}.multiselect__option--group{background:#ededed;color:#35495e}.multiselect__option--group.multiselect__option--highlight{background:#35495e;color:#fff}.multiselect__option--group.multiselect__option--highlight:after{background:#35495e}.multiselect__option--disabled.multiselect__option--highlight{background:#dedede}.multiselect__option--group-selected.multiselect__option--highlight{background:#ff6a6a;color:#fff}.multiselect__option--group-selected.multiselect__option--highlight:after{background:#ff6a6a;content:attr(data-deselect);color:#fff}.multiselect-enter-active,.multiselect-leave-active{transition:all .15s ease}.multiselect-enter,.multiselect-leave-active{opacity:0}.multiselect__strong{margin-bottom:8px;line-height:20px;display:inline-block;vertical-align:top}[dir=rtl] .multiselect{text-align:right}[dir=rtl] .multiselect__select{right:auto;left:1px}[dir=rtl] .multiselect__tags{padding:8px 8px 0 40px}[dir=rtl] .multiselect__content{text-align:right}[dir=rtl] .multiselect__option:after{right:auto;left:0}[dir=rtl] .multiselect__clear{right:auto;left:12px}[dir=rtl] .multiselect__spinner{right:auto;left:1px}@keyframes a{0%{transform:rotate(0)}to{transform:rotate(2turn)}}
```
Comment (szafran89) [inline view/frontend/web/js/assets/vue-multiselect.min.css]: There is no need to add styles to repo

File: view/frontend/web/js/index.vue (hunk 1 above)
Comment (szafran89) [reply view/frontend/web/js/index.vue]: Change it to `<style src="vue-multiselect/dist/vue-multiselect.min.css">`

Comment (kjugi) [reply view/frontend/web/js/index.vue]: moved and changed

File: view/frontend/web/js/assets/vue-multiselect.min.css (hunk 4 above)
Comment (kjugi) [reply view/frontend/web/js/assets/vue-multiselect.min.css]: ok, right

File: view/frontend/web/js/components/steps/ThePaymentStep.vue (hunk 3 above)
Comment (kjugi) [reply view/frontend/web/js/components/steps/ThePaymentStep.vue]: i'm using it b/c track-by attribute where i can provide id of option not working as I expected b/c it's adding whole object where we need only value from this object and watchers can do this for me.

File: view/frontend/web/js/components/steps/TheShippingStep.vue (hunk 2 above)
Comment (kjugi) [reply view/frontend/web/js/components/steps/TheShippingStep.vue]: same as in payment step

**Comment by szafran89** [comment]
@kjugi FYI there are new commits in PR

**Pull request merged**